pynest/examples/if_curve.py

```diff
@@ -84,12 +84,9 @@ def build(self):
         nest.SetKernelStatus({'local_num_threads': self.n_threads})
 
         #######################################################################
-        # We set the default parameters of the neuron model to those
-        # defined above and create neurons and devices.
+        # We create neurons and devices with specified parameters.
 
-        if self.params:
-            nest.SetDefaults(self.model, self.params)
-        self.neuron = nest.Create(self.model, self.n_neurons)
+        self.neuron = nest.Create(self.model, self.n_neurons, self.params if self.params else {})
```

Comment:
`Create` has default `params=None`, so it should be possible to do
```
self.neuron = nest.Create(self.model, self.n_neurons, self.params)
```
here?

Reply:
yes, the post-fix if-else construct in Python always seems enormously unreadable. I know it basically covers the ternary `<>?<>:<>` known from many languages, but it's unnecessary in most cases.

```suggestion
        self.neuron = nest.Create(self.model, self.n_neurons, self.params)
```

Here the `__init__()` above just needs to be changed too. That's what caused the problem here.